src/services/db/GitFileCommitService.ts

```diff
@@ -14,14 +16,34 @@ import GitFileSystemService from "./GitFileSystemService"
  * 2. Creates non-asset related commits to staging-lite
  */
 export default class GitFileCommitService {
-  private readonly STAGING_LITE_BRANCH = "staging-lite"
-
   private readonly gitFileSystemService: GitFileSystemService
 
   constructor(gitFileSystemService: GitFileSystemService) {
     this.gitFileSystemService = gitFileSystemService
   }
 
+  async pushToGithub(
+    sessionData: UserWithSiteSessionData,
+    shouldUpdateStagingLite: boolean
+  ) {
+    // We await the push to staging FIRST, and then push to staging-lite
+    // We don't want a case when staging lite updates but staging doesn't
+    const res = this.gitFileSystemService.push(
+      sessionData.siteName,
+      STAGING_BRANCH
+    )
+
+    if (shouldUpdateStagingLite) {
+      res.andThen(() =>
+        this.gitFileSystemService.push(
+          sessionData.siteName,
+          STAGING_LITE_BRANCH
+        )
```

Comment:
issue: would this not create a separate chain? The `res` object will not be updated with this new call when we await later.

Reply:
2e06bec48fd39e96336769aa4ca9b5b70ac81d19
for avoidance of doubt, one backoff per promise -> ResultAsyncs are converted to native Promise<T> to be compatible with `exponential-backoff`